src/View/mainpage/MainPage.py

```diff
@@ -263,3 +279,67 @@ def format_data(self, size):
+            self.left_panel.addTab(self.structures_tab, "Structures")
+        elif hasattr(self, 'structures_tab'):
+            del self.structures_tab
+
+        if moving_dict_container.has_modality("rtdose"):
+            self.isodoses_tab = IsodoseTab()
+            self.isodoses_tab.request_update_isodoses.connect(self.update_views)
+            self.left_panel.addTab(self.isodoses_tab, "Isodoses")
+        elif hasattr(self, 'isodoses_tab'):
+            del self.isodoses_tab
+
+        # Hide left panel if no rtss or rtdose
+        if not moving_dict_container.has_modality("rtss") and not moving_dict_container.has_modality("rtdose"):
+            self.left_panel.hide()
+
+        roi_color_dict = self.structures_tab.color_dict if hasattr(self, 'structures_tab') else None
+        iso_color_dict = self.isodoses_tab.color_dict if hasattr(self, 'isodoses_tab') else None
+
+        # Add it to the panel
+        # Create an Image Fusion View containing single-slice and 3-slice views
+        # Add structures tab to left panel
+        
+        self.image_fusion_view = QStackedWidget()
+        
+        print("Create Image Fusion Tab")
+        self.image_fusion_view_axial = ImageFusionView(format_metadata=False, color=True)
+        self.image_fusion_view_sagittal = ImageFusionView(slice_view="sagittal", color=True)
+        self.image_fusion_view_coronal = ImageFusionView(slice_view="coronal", color=True)
+        
+        # Rescale the size of the scenes inside the 3-slice views
+        self.image_fusion_view_axial.zoom = 0.5
+        self.image_fusion_view_sagittal.zoom = 0.5
+        self.image_fusion_view_coronal.zoom = 0.5
+        self.image_fusion_view_axial.update_view(zoom_change = True, color=True)
+        self.image_fusion_view_sagittal.update_view(zoom_change=True, color=True)
+        self.image_fusion_view_coronal.update_view(zoom_change=True, color=True)
+
+        self.image_fusion_four_views = QWidget()
+        self.image_fusion_four_views_layout = QGridLayout()
+        for i in range(2):
+            self.image_fusion_four_views_layout.setColumnStretch(i, 1)
+            self.image_fusion_four_views_layout.setRowStretch(i, 1)
+        self.image_fusion_four_views_layout.addWidget(self.image_fusion_view_axial, 0, 0)
+        self.image_fusion_four_views_layout.addWidget(self.image_fusion_view_sagittal, 0, 1)
+        self.image_fusion_four_views_layout.addWidget(self.image_fusion_view_coronal, 1, 0)
+        self.image_fusion_four_views.setLayout(self.image_fusion_four_views_layout)
+        
+        self.image_fusion_single_view = ImageFusionView(color=True)
+        
+        self.image_fusion_view.addWidget(self.image_fusion_four_views)
+        self.image_fusion_view.addWidget(self.image_fusion_single_view)
+        self.image_fusion_view.setCurrentWidget(self.image_fusion_four_views)
+        
+
+        # Add Image Fusion Tab
+        self.right_panel.addTab(self.image_fusion_view, "Image Fusion")
```

Comment:
Same here, a lot of the code is similar to setup_central_widget(), so can you reuse the code, make one mutual function and call it from there?

Reply:
From an attempt to refactor this code, I believe with the concepts I've learnt from SE design - that it's best to leave this function as a standalone function. A few reasons why:
- This will cause the function to be very long and maintain 2 fairly different use cases
- Keeps low coupling and maintain cohesion otherwise merged the two functions together will cause counterintuitive effects
- This will maintain the upgradability of this function as future implementations (manual fusion) may modify the functionality.